File: Heartcooking.API/Models/Product.cs
using System.Collections.Generic;

namespace Heartcooking.API.Models
{
    public class Product : BaseEntity
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public double Carbohydrates { get; set; }
        public double Fat { get; set; }
        public double Fiber { get; set; }
        public double Kilocalories { get; set; }
        public double Protein { get; set; }
        public double Salt { get; set; }
        public double SaturatedFat { get; set; }
        public double Sugar { get; set; }
        public bool IsVegan { get; set; }

        public ICollection<Allergen> Allergens { get; set; }
    }
}

File: Heartcooking-SPA/src/app/models/product.ts
import { Allergen } from './allergen';
import { BaseModel } from './base-model';
import { Photo } from './photo';

export interface Product extends BaseModel {

  name: string;
  description: string;
  carbohydrates: number;
  fat: number;
  fiber: number;
  kilocalories: number;
  protein: number;
  salt: number;
  saturatedFat: number;
  suger: number;
  isVegan: boolean;
  photoUrl: string;
  photos?: Photo[];
  allergens?: Allergen[];
}
